File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/one/Long_ScreenOne_LastBarHigher.java
package com.kishlaly.ta.analyze.tasks.blocks.one;

import com.kishlaly.ta.model.BlockResultJava;
import com.kishlaly.ta.model.SymbolDataJava;
import com.kishlaly.ta.utils.LogJava;

import static com.kishlaly.ta.model.BlockResultCodeJava.LAST_QUOTES_NOT_ASCENDING_SCREEN_1;
import static com.kishlaly.ta.model.BlockResultCodeJava.OK;

/**
 * the last quote is higher than the penultimate one
 */
public class Long_ScreenOne_LastBarHigher implements ScreenOneBlockJava {
    @Override
    public BlockResultJava check(SymbolDataJava screen) {
        boolean lastBarHigher = screen.getLastQuote().getLow() > screen.getPreLastQuote().getLow()
                && screen.getLastQuote().getHigh() > screen.getPreLastQuote().getHigh();

        if (!lastBarHigher) {
            LogJava.recordCode(LAST_QUOTES_NOT_ASCENDING_SCREEN_1, screen);
            LogJava.addDebugLine("The last bar is not higher than the penultimate on the long-term screen");
            return new BlockResultJava(screen.getPreLastQuote(), LAST_QUOTES_NOT_ASCENDING_SCREEN_1);
        }
        return new BlockResultJava(screen.getPreLastQuote(), OK);
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/blocks/one/Long_ScreenOne_LastBarHigher.kt
package com.kishlaly.ta.analyze.tasks.blocks.one

import com.kishlaly.ta.model.BlockResult
import com.kishlaly.ta.model.BlockResultCode
import com.kishlaly.ta.model.SymbolData
import com.kishlaly.ta.utils.Log

/**
 * the last quote is higher than the penultimate one
 */
class Long_ScreenOne_LastBarHigher : ScreenOneBlock {
    override fun check(screen: SymbolData): BlockResult {
        val lastBarHigher = (screen.lastQuote.low > screen.preLastQuote.low
                && screen.lastQuote.high > screen.preLastQuote.high)

        if (!lastBarHigher) {
            Log.recordCode(BlockResultCode.LAST_QUOTES_NOT_ASCENDING_SCREEN_1, screen)
            Log.addDebugLine("The last bar is not higher than the penultimate on the long-term screen")
            return BlockResult(screen.lastQuote, BlockResultCode.LAST_QUOTES_NOT_ASCENDING_SCREEN_1)
        }
        return BlockResult(screen.lastQuote, BlockResultCode.OK)
    }
}